checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1602 @@
+        }
+      } else {
+        doGenKillForPseudoAssignment(node, facts, lhsVar, rhs);
+      }
+    }
+  }
+
+  /**
+   * Remove any facts from a fact set that contain a {@link LocalVarWithTree} {@code var}.
+   *
+   * @param facts the set of facts
+   * @param var a variable
+   */
+  private void removeFactContainingVar(
+      Set<ImmutableSet<LocalVarWithTree>> facts, LocalVariableNode var) {
+    Set<LocalVarWithTree> setContainingRhs = getFactContainingVar(facts, var);
+    facts.remove(setContainingRhs);
+  }
+
+  /**
+   * Remove any {@link TypeCastNode}s wrapping a node, returning the operand nested within the type
+   * casts.
+   *
+   * @param node a node
+   * @return node, but with any surrounding typecasts removed
+   */
+  private Node removeCasts(Node node) {
+    while (node instanceof TypeCastNode) {
+      node = ((TypeCastNode) node).getOperand();
+    }
+    return node;
+  }
+
+  /**
+   * Update a set of tracked definitions to account for a (pseudo-)assignment to some variable, as
+   * in a gen-kill dataflow analysis problem. Pseudo-assignments may include operations that
+   * "assign" to a temporary variable. E.g., for an expression {@code b ? x : y}, this method may
+   * process an "assignment" from {@code x} or {@code y} to the temporary variable representing the
+   * ternary expression.
+   *
+   * @param node the node performing the pseudo-assignment; it is not necessarily an assignment node
+   * @param facts the tracked facts
+   * @param lhsVar the left-hand side variable for the pseudo-assignment
+   * @param rhs the right-hand side for the pseudo-assignment
+   */
+  private void doGenKillForPseudoAssignment(
+      Node node, Set<ImmutableSet<LocalVarWithTree>> facts, LocalVariableNode lhsVar, Node rhs) {
+    // Replacements to eventually perform in the facts.  We keep this map to avoid a
+    // ConcurrentModificationException in the loop below.
+    Map<ImmutableSet<LocalVarWithTree>, ImmutableSet<LocalVarWithTree>> replacements =
+        new LinkedHashMap<>();
+    // construct lhsVarWithTreeToGen once outside the loop for efficiency
+    LocalVarWithTree lhsVarWithTreeToGen =
+        new LocalVarWithTree(new LocalVariable(lhsVar), node.getTree());
+    for (ImmutableSet<LocalVarWithTree> fact : facts) {
+      Set<LocalVarWithTree> kill = new LinkedHashSet<>();
+      // always kill the lhs var if present
+      addLocalVarWithTreeToSetIfPresent(fact, lhsVar.getElement(), kill);
+      LocalVarWithTree gen = null;
+      // if rhs is a variable tracked in the fact, gen the lhs
+      if (rhs instanceof LocalVariableNode) {
+        LocalVariableNode rhsVar = (LocalVariableNode) rhs;
+        for (LocalVarWithTree lvwt : fact) {
+          if (lvwt.localVar.getElement().equals(rhsVar.getElement())) {
+            gen = lhsVarWithTreeToGen;
+            // We remove temp vars from tracking once they are assigned to another location.
```

Comment:
I believe this is just an optimization to improve error messages: if a temporary is assigned to another location (such as a local or a field), we'd rather issue a warning about the local/field that's actually in the source code than the temporary.

As for temporary variables generally, I'm not sure what to say about them. We create them for expressions so that those expressions can have their types refined by the relevant transfer functions, even if they're not traditionally kept in a CFStore. I added a note to this effect to the class documentation, but IMO it feels like a non-sequitur.